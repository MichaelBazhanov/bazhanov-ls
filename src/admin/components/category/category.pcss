@import "../../../styles/mixins.pcss";

.category-component {
	height: 100%;
}
.bottom-line {
	padding-top: 70px;
	margin-top: auto;
	padding-left: 25%;

	@include desktopHd {
	}
	@include desktop {
		padding-left: 0;
	}
	@include tablets {
	}
	@include phones {
	}
}
.item {
	margin-bottom: 20px;
	&:last-child {
		margin-bottom: 0;
	}
}