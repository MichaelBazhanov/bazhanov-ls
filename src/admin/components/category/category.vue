<template>
	<div class="category-component">
		<card slim>
			<edit-line
				slot="title"
				v-model="categoryTitle"
				:editModeByDefault="empty"
				@remove="$emit('remove', $event)"
			/>
			<template slot="content">
				<ul class="skills" v-if="empty == false">
					<li class="item" v-for="skill in skills" :key="skill.id">
						<skill :skill="skill"/>
					</li>
				</ul>
				<div class="bottom-line">
					<skill-add-line :blocked="empty"/>
				</div>
			</template>
		</card>
	</div>
</template>

<script>
import card from "../Card";
import editLine from "../editLine";
import skill from "../skill";
import skillAddLine from "../skillAddLine";

export default {
	props: {
		empty: Boolean,
		title: {
			type: String,
			default: ''
		},
		skills: {
			type: Array,
			dafault: ()=>[]
		}
	},
	components: {
		card,
		editLine,
		skill,
		skillAddLine,
	},
	data() {
		return {
			categoryTitle: this.title,
		}
	},
}
</script>

<style lang="postcss" src="./category.pcss" scoped></style>
