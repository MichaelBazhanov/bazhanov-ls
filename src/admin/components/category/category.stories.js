import category from "./category.vue";
import { action } from "@storybook/addon-actions";

const methods = {
	onRemove: action("onRemove"),
	onRemoveSkill: action("onRemoveSkill"),
	onEditSkill: action("onEditSkill")
};

export default {
	title: "category",
	components: {category}
};

//необходимые данные для отображения
const skills = [
	{ "id": 0, "title": "HTMl", "percent": 80 },
	{ "id": 2, "title": "HTMl2", "percent": 20 },
	{ "id": 3, "title": "HTMl3", "percent": 10 }
];

export const defaultView = () => ({
	components: {category},
	template: `
		<category
			:title="title"
			:skills="skills"
			@remove="onRemove"
			@remove-skill="onRemoveSkill"
			@edit-skill="onEditSkill"
		/>
	`,
	data() {
		return {
			title: "Frontend",
			skills
		}
	},
	methods
});

defaultView.story = {
	name: "Стандартный вид",
};
