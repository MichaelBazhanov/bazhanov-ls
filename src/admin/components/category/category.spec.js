import category from './category'
import { mount } from "@vue/test-utils"

it('Category snapshot', async () => {
	//steps
	const wrapper = mount(category)

	//actions
	//check
	expect(wrapper.element).toMatchSnapshot()
})