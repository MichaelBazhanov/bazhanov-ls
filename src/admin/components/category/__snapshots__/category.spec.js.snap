// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category snapshot 1`] = `
<div
  class="category-component"
>
  <div
    class="card-component"
  >
    <div
      class="header header_plain"
    >
      <div
        class="text"
      />
       
      <div
        class="edit-line-component"
      >
        <div
          class="title"
        >
          <div
            class="text"
          >
            
          </div>
           
          <div
            class="icon"
          >
            <!---->
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="content"
    >
      <ul
        class="skills"
      />
       
      <div
        class="bottom-line"
      >
        <div
          class="skill-add-line-component"
        >
          <label
            class="input skill-text"
          >
            <!---->
             
            <input
              class="input__elem field__elem"
              placeholder="Новый навык"
            />
             
            <span
              class="percent-title"
            >
              %
            </span>
             
            <div
              class="input__error-tooltip"
            >
              <!---->
            </div>
          </label>
           
          <label
            class="input skill-percent percent no-side-paddings"
          >
            <!---->
             
            <input
              class="input__elem field__elem"
              max="100"
              maxlength="2"
              min="0"
              type="number"
            />
             
            <span
              class="percent-title"
            >
              %
            </span>
             
            <div
              class="input__error-tooltip"
            >
              <!---->
            </div>
          </label>
           
          <!---->
        </div>
      </div>
    </div>
  </div>
</div>
`;
